packages/logging/service_test.go

```diff
@@ -324,3 +340,199 @@ func Test_QueryForDogu(t *testing.T) {
 		assert.ErrorContains(t, err, assert.AnError.Error())
 	})
 }
+
+func TestLoggingService_SetLogLevel(t *testing.T) {
+	tests := []struct {
+		name          string
+		req           *pb.LogLevelRequest
+		xResponse     bool
+		xResponseCode codes.Code
+	}{
+		{
+			name: "Set LogLevel DEBUG",
+			req: &pb.LogLevelRequest{
+				DoguName: "test",
+				LogLevel: pb.LogLevel_DEBUG,
+			},
+			xResponse:     true,
+			xResponseCode: codes.OK,
+		},
+		{
+			name: "Set LogLevel INFO",
+			req: &pb.LogLevelRequest{
+				DoguName: "test",
+				LogLevel: pb.LogLevel_INFO,
+			},
+			xResponse:     true,
+			xResponseCode: codes.OK,
+		},
+		{
+			name: "Set LogLevel WARN",
+			req: &pb.LogLevelRequest{
+				DoguName: "test",
+				LogLevel: pb.LogLevel_WARN,
+			},
+			xResponse:     true,
+			xResponseCode: codes.OK,
+		},
+		{
+			name: "Set LogLevel ERROR",
+			req: &pb.LogLevelRequest{
+				DoguName: "test",
+				LogLevel: pb.LogLevel_ERROR,
+			},
+			xResponse:     true,
+			xResponseCode: codes.OK,
+		},
+		{
+			name: "Set wrong LogLevel",
+			req: &pb.LogLevelRequest{
+				DoguName: "test",
+				LogLevel: 4,
+			},
+			xResponse:     false,
+			xResponseCode: codes.InvalidArgument,
+		},
+		{
+			name: "Empty dogu name",
+			req: &pb.LogLevelRequest{
+				DoguName: "",
+				LogLevel: pb.LogLevel_DEBUG,
+			},
+			xResponse:     false,
+			xResponseCode: codes.InvalidArgument,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			mockedLogProvider := newMockLogProvider(t)
+			mockedConfigProvider := newMockConfigProvider(t)
+			mockedDoguRestarter := newMockDoguRestarter(t)
+			mockedConfigurationContext := NewMockConfigurationContext(t)
+
+			if tc.xResponse {
+				mockedConfigProvider.On("DoguConfig", mock.Anything).Return(mockedConfigurationContext)
+
+				mockedConfigurationContext.On("Get", mock.Anything).Return("", nil)
+				mockedConfigurationContext.On("Set", mock.Anything, mock.Anything).Return(nil)
+
+				mockedDoguRestarter.On("RestartDogu", mock.Anything, mock.Anything).Return(nil)
```

Comment:
You alread wrote tests with `mock.EXPECT().YourMethodHere()` logic which have a better refactorability.

What is the reason you don't do this here?

If this could be replaced with the EXPECT clause then probably other `.On("function name")` too in this test file.